browser/src/index.ts

```diff
@@ -0,0 +1,103 @@
+
+function unpackU64toU8(n: BigInt): Uint8Array {
+    //@ts-ignore
+    return new Uint8Array(new BigInt64Array([n]).buffer)
+}
+
+function packU8toU64(bytes: Uint8Array): BigInt {
+    return (new BigInt64Array(bytes.buffer))[0]
+}
+
+function stringToBytes(s: string, offset: number, len: number): Uint8Array {
+    return new Uint8Array(s.slice(offset, len).split("").map(c => c.charCodeAt(0)))
+}
+
+function makeEnv(plugin: ExtismPlugin): any {
+    return {
+        extism_alloc(n: BigInt): BigInt {
```

Comment:
This only works if the plugin only uses extism_alloc once, I think the http example might fail. I can help port some of the allocation logic over from the Rust runtime.